### ISSUE TITLE
Test Explorer is not running my tests correctly (but when Adaptor is restored to NUnit3TestAdapter version 3.15.1 it works OK)

### ISSUE THREAD

**Issue by foxy-tester** [nunit/nunit3-vs-adapter#702]
Required Info:
* NUnit version 3.12.0 and NUnit3TestAdapter version NUnit3TestAdapter 3.16.0
* Microsoft Visual Studio Professional 2019
Version 16.4.2. VisualStudio.16.Release/16.4.2+29613.14, with Microsoft .NET Framework
Version 4.8.03752

Issue:
I’m use Test Explorer to run my NUnit Tests and when I select an individual test to run then the is result is all tests are run. When I select all tests to run, then none run.  

I have several other issues with Test Explorer, but this is the main problem because it makes Test Explorer virtually unusable and I’m finding it difficult to just get all my tests run when some should not be run at the same time. I’m not using a settings file and not running in parallel mode.

Possibly a different or related issue: but old tests (no long part of the code) are not cleared out even after a clean and rebuild. These appear under 'Project Unknown'. They never get cleared down so I assume they are stored somewhere like c:/%TEMP%/. Notably, though old test results never get cleared down. Also raised with Microsoft against Test Explorer here: https://developercommunity.visualstudio.com/content/problem/878461/test-explorer-is-picking-random-tests-to-run-rathe.html?childToView=881238#comment-881238

**Comment by OsirisTerje** [comment]
First part is a duplicate of #690, and is fixed in 3.16.1, to be released in a couple of days. 
Second part is a valid bug, and is also solved in 3.16.1.

**Comment by OsirisTerje** [comment]
@foxy-tester  Please verify if this beta version works in your case https://www.myget.org/feed/nunit/package/nuget/NUnit3TestAdapter/3.16.1-dev-01246

**Comment by OsirisTerje** [comment]
This issue is resolved by Version 3.16.1 which is released now on [nuget.org](https://www.nuget.org/packages/NUnit3TestAdapter/3.16.1) and VSIX on [Visual Studio marketplace](https://marketplace.visualstudio.com/items?itemName=NUnitDevelopers.NUnit3TestAdapter).  See [all issues resolved](https://github.com/nunit/nunit3-vs-adapter/issues?q=is%3Aissue+milestone%3A3.16.1+is%3Aclosed) and  [Release notes](https://github.com/nunit/docs/wiki/Adapter-Release-Notes) for details.

**Comment by foxy-tester** [comment]
Unfortunately, the 2nd part is not resolved. Namely: 'old tests (no long part of the code) are not cleared out even after a clean and rebuild. These appear under 'Project Unknown'. They never get cleared down so I assume they are stored somewhere like c:/%TEMP%/. Notably, though old test results never get cleared down.'  The old test results and tests listed under <Unknown Project> for some reason are not getting cleared at all. I did a Clean and Rebuild again & re-ran my tests but this remains an issue.  What were duplicate tests are not just listed under 'Unknown Project> so perhaps there is a way to clear this data manually once and for all. Please advise.

I can see that the duplication of tests within the groups no longer happens though. What is also still an issue is that when I select several groups to run, all the tests in the namespace are run. So the selection of a sub-set of tests is not working correctly. Though a smaller sub-set selection did work for me. So its a bit random but not really fixed.

**Comment by OsirisTerje** [comment]
@foxy-tester Can you provide a small repro for this (2nd part) ?  
(Need to raise a new issue though for this part, but we can do that afterwards)

**Comment by foxy-tester** [comment]
Will do so shortly. Our system is huge, so will need to put something together just to show this issue. However, it would be good to be sure old results still remaining is not just a 'hang-over' from the previous issue which is fixed and if you know where to clean up the files (as I presume its not within the project /bin folder or would have gone by now)

**Comment by OsirisTerje** [comment]
I am not sure, and haven't got any clear answers from MS about this yet.  I assume it is somewhere in the .vs folder, and as you have noted, it doesn't seems to be cleared by clean nor rebuild.
